KakapoExample/KakapoTests/RouterTests.swift

```diff
@@ -0,0 +1,720 @@
+                expect(responseArray?.lastObject?["id"]).toEventually(equal(19))
+            }
+            
+            it("should return nil for objects not serializable to JSON") {
+                router.get("/nothing/:id"){ request in
+                    return Optional.Some("none")
+                }
+                
+                var called = false
+                NSURLSession.sharedSession().dataTaskWithURL(NSURL(string: "http://www.test.com/nothing/1")!) { (data, response, _) in
+                    called = true
+                    expect(data?.length).to(equal(0))
+                }.resume()
+                
+                expect(called).toEventually(beTrue())
+            }
+        }
+        
+        describe("Multiple Routers") {
+            var router: Router!
+            
+            it("Should handle multiple Routers that register differents urls") {
+                router = Router.register("http://www.test.com")
+                
+                var info: URLInfo? = nil
+                var responseURL: NSURL? = nil
+                
+                var secondInfo: URLInfo? = nil
+                var secondResponseURL: NSURL? = nil
+                let secondRouter = Router.register("www.host2.com")
+                
+                router.get("/users/:id") { request in
+                    info = (components: request.components, queryParameters: request.queryParameters)
+                    return nil
+                }
+                
+                secondRouter.get("/messages/:user") { request in
+                    secondInfo = (components: request.components, queryParameters: request.queryParameters)
+                    return nil
+                }
+                
+                NSURLSession.sharedSession().dataTaskWithURL(NSURL(string: "http://www.test.com/users/1")!) { (data, response, _) in
+                    responseURL = response?.URL
+                    }.resume()
+                
+                NSURLSession.sharedSession().dataTaskWithURL(NSURL(string: "http://www.host2.com/messages/24")!) { (data, response, _) in
+                    secondResponseURL = response?.URL
+                    }.resume()
+                
+                expect(info?.components).toEventually(equal(["id" : "1"]))
+                expect(info?.queryParameters).toEventually(equal([]))
+                expect(responseURL?.absoluteString).toEventually(equal("http://www.test.com/users/1"))
+                
+                expect(secondInfo?.components).toEventually(equal(["user" : "24"]))
+                expect(secondInfo?.queryParameters).toEventually(equal([]))
+                expect(secondResponseURL?.absoluteString).toEventually(equal("http://www.host2.com/messages/24"))
+            }
+            
+            it ("Should not handle invalid registrations on routers (leading and trailing slashes should match the absolute request)") {
```

Comment:
O RLY? so if you miss the trailing won't work? BTW these tests should be in the URL matcher not in the server.

Server: test if matched urls callback
Decomposer: test the matcher